data/events.txt

```diff
@@ -1074,107 +1054,20 @@ event "fw safe passage ends"
 
 
 event "navy using mark ii ships"
+	variant "Rainmaker"
+		add "Rainmaker (Mark II)" 3
+	variant "Gunboat"
+		add "Gunboat (Mark II)" 3
+	variant "Frigate"
+		add "Frigate (Mark II)" 3
+	variant "Cruiser"
+		add "Cruiser (Mark II)" 3
+	variant "Carrier"
+		add "Carrier (Mark II)" 3
```

Comment:
Something that I think is important pointing out is how the inclusion of this new variant functionality differs from the old system of adding new fleet variants. Whereas before 3/4ths of all Navy fleets became Mark II fleets from this event, this change instead makes it so that 3/4ths of all Navy ships within a single fleet variant are Mark II. That is, you no longer get fleets of all Mark II ships and all Mark I ships, but fleets of mostly Mark II ships and some Mark I ships.

Is this a desirable outcome for Navy fleets? Is there anywhere else that comes to mind where this mixing of ship variants could be an issue?

Reply:
The mixed fleet looks fine to me, but I guess a remove would also be in order. For example to remove "Mark I"s after some time.

Reply:
And we should test the game how it handles empty variants (for if some content creator makes a mistake and removes all variants).